### DISCUSSION TITLE
Mocking CRUD with integer ids

### DISCUSSION THREAD

**Discussion by peternormann** [mockoon/mockoon#1368] [Help]
Hi,

I have spent tons of hours in frustration trying to create basic crud endpoints where the id of the entities are integers, rather than uuid.

I want a POST request with data (that does not contain an id) to get populated with an incremented integer as id and added to the data bucket, so that I can serve that as response.

I am sure I am just being dense and have missed something basic, and this belief leads me to revisit mockoon at every major release only to find myself wasting hours in rabbit holes trying to get it to work in vain.

Does anyone have some pointers?

**Comment by 255kb** [comment]
Hi, 
While CRUD routes work with integers, POST routes create objects with a string UUID, unless you provide the id property. There is currently no automated mechanism to check the existing elements and act accordingly.
Feel free to open an issue, it would be a good improvement.

**Reply by 255kb** [reply]
Mockoon is an open-source project and relies on your support. If you found this answer helpful, please consider [sponsoring us](https://github.com/sponsors/mockoon). Additionally, we're in the process of developing a [SaaS platform](https://mockoon.com/pro/) to ensure the project's long-term sustainability. Currently, our platform offers an AI assistant, data synchronization, and team collaboration features, with cloud deployments coming soon. Thank you for your support!
